web/main/models.py

```diff
@@ -1071,20 +1076,34 @@ def casebook_fts(
                 result_id__in=textblock_ids
             )
 
+        query_vector: Union[SearchQuery, str]
+        if query_str:
+            query_vector = SearchQuery(query_str, config="english")
+        else:
+            query_vector = ""
+
         if query_vector:
             base_query = base_query.filter(document=query_vector)
+
         results = base_query.filter(category=category)
         results = results.annotate(rank=SearchRank(F("document"), query_vector))
         display_name = get_display_name_field(category)
         results = results.order_by("-rank", display_name)
-        results = Paginator(results, page_size).get_page(page)
 
-        ids = sorted([r.result_id for r in results])
-        query_class = ({"legal_doc_fulltext": LegalDocument, "textblock": TextBlock, "link": Link})[
-            category
-        ]
+        results_chunk = Paginator(results, page_size).get_page(page)
+        ids = sorted([r.result_id for r in results_chunk])
+
+        # Can replace w/ match statement when upgraded to 3.10
+        query_class: ResourceType
+        if category == "legal_doc_fulltext":
+            query_class = LegalDocument
+        elif category == "textblock":
+            query_class = TextBlock
+        elif category == "link":
+            query_class = Link
+
         content_name = "description" if category == "link" else "content"
-        ids_headlines = (
+        ids_headlines_query = (
```

Comment:
> Is this variable renamed with the appendix `_query` because it's assigned to a `queryset`? 

Yeah, mypy really does not like variables changing types.